spec/omniauth/auth0/jwt_validator_spec.rb

```diff
@@ -225,21 +257,26 @@
         kid: jwks_kid
       }
       token = make_rs256_token(payload)
-      decoded_token = make_jwt_validator(domain).decode(token)
+      decoded_token = make_jwt_validator(opt_domain: domain).decode(token)
       expect(decoded_token.first['sub']).to eq(sub)
     end
   end
 
   private
 
-  def make_jwt_validator(opt_domain = domain)
-    OmniAuth::Auth0::JWTValidator.new(
-      Options.new(
-        opt_domain,
-        client_id,
-        client_secret
-      )
-    )
+  def make_jwt_validator(opt_domain: domain, opt_issuer: nil)
+    opts = if opt_issuer.nil?
```

Comment:
You already have as part of the `initialize` method of the verifier a check for the presence of the `issuer` value. 

```
if options.respond_to?(:issuer)
```

Can't you instantiate it directly using named arguments? Like you did here https://github.com/auth0/omniauth-auth0/pull/77/files#diff-708c8cee2bf0dc225b55a0c36afaa97dR260. Why do you need to use `Options` and `OptionsWithIssuer`?

Reply:
Ah, didn't realize `Options` was a `Struct` and not an `OpenStruct` before and was confused why I couldn't add values, which is why I made a different struct. Switching the implementation to use an `OpenStruct` where I add the issuer if it's there. Can't pass in the issuer if it's nil because the OpenStruct/Struct will return true for the `respond_to?` method, even though the value it returns is actually nil.